### PR TITLE
Type annotations & code refactor

### PR DESCRIPTION
**Pull request by DanielAvdar**
Type annotations contribute significantly to code readability, facilitate code completion, and highlight type incompatibility issues. Please wait for checks to complete to ensure annotations are compatible with all Python versions.

### CODE REVIEW

File: supervised/algorithms/algorithm.py [hunk 1]
```diff
@@ -38,15 +40,15 @@ def reload(self):
...
-        self,
-        X,
-        y,
-        sample_weight=None,
-        X_validation=None,
-        y_validation=None,
-        sample_weight_validation=None,
-        log_to_file=None,
-        max_time=None,
+            self,
```
Comment (pplonski) [inline supervised/algorithms/algorithm.py:43]: why are you adding spaces?

File: supervised/algorithms/algorithm.py [hunk 2]
```diff
@@ -76,18 +78,18 @@ def get_fname(self):
...
-        X_validation,
-        y_validation,
-        model_file_path,
-        learner_name,
-        target_name=None,
-        class_names=None,
-        metric_name=None,
-        ml_task=None,
-        explain_level=2,
+            self,
```
Comment (pplonski) [inline supervised/algorithms/algorithm.py:81]: spaces again?

File: supervised/automl.py [hunk 3]
```diff
@@ -403,44 +397,44 @@ def __init__(
...
-        self,
-        X: Union[numpy.ndarray, pandas.DataFrame],
-        y: Union[numpy.ndarray, pandas.Series],
-        sample_weight: Optional[Union[numpy.ndarray, pandas.Series]] = None,
-        cv: Optional[Union[Iterable, List]] = None,
-        sensitive_features: Optional[
-            Union[numpy.ndarray, pandas.Series, pandas.DataFrame]
-        ] = None,
+            self,
+            X: Union[np.ndarray, pd.DataFrame],
```
Comment (pplonski) [inline supervised/automl.py:401]: why you changed `numpy` to `np` and `pandas` to `pd`?

**Review by pplonski — CHANGES_REQUESTED** [review]
Thank you for PR! I think some of the formatting is not needed and changes in packages names.

File: supervised/automl.py (hunk 3 above)
Comment (DanielAvdar) [reply supervised/automl.py:401]: These aliases have become standardized within the Python data science community, so you'll often see them used in tutorials, documentation, and code examples. It's not a mandatory change, but it's a widely accepted convention that helps improve code readability and efficiency.

File: supervised/algorithms/algorithm.py (hunk 1 above)
Comment (DanielAvdar) [reply supervised/algorithms/algorithm.py:43]: This is automatic reformatting by the IDE.

**Comment by DanielAvdar** [comment]
I will open a new PR with the desired changes. I'd like to emphasize that it's possible to enforce code styling using pylint. Furthermore, we can add a code style check to the PR. Is that something you would like @pplonski ?

**Pull request closed without merging**